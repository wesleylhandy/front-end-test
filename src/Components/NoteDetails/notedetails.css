.NoteDetails {
	padding: 1rem;
}
