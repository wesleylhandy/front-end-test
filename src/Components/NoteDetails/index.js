import React from "react";
import Button from "../Button";
import PropTypes from "prop-types";
import "./notedetails.css";

const NoteDetails = ({ note = {}, toggleReadStatus }) => {
	const { subject, body, read } = note;
	return subject && body && typeof read === "boolean" ? (
		<section className="NoteDetails">
			<h3 className="NoteDetails__title">{subject}</h3>
			<p className="NoteDetails__subject">{body}</p>
			<Button
				className="NoteDetails__button"
				clickHandler={toggleReadStatus}
				text={`Mark as ${read ? "unread" : "read"}`}
			/>
		</section>
	) : null;
};

NoteDetails.propTypes = {
	note: PropTypes.shape({
		subject: PropTypes.string,
		body: PropTypes.string,
		read: PropTypes.bool,
	}),
	toggleReadStatus: PropTypes.func.isRequired,
};

export default NoteDetails;
