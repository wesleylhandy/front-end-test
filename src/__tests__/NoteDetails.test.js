import React from "react";
import NoteDetails from "../Components/NoteDetails";
import { create, act } from "react-test-renderer";

const note = {
	subject: "Lorem ipsum dolor sit amet",
	body:
		"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non velit vehicula lacus dictum tristique. Quisque ac justo magna. Duis eu est purus. Quisque ut enim tortor. Morbi interdum risus a lorem posuere dapibus. Morbi orci est, varius sed dolor quis, efficitur scelerisque sapien. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Aenean at libero sapien. Morbi ultrices malesuada nisl, sit amet aliquam ipsum egestas non. Nunc at massa vel metus vestibulum fringilla in vitae est. Pellentesque eget lorem eget arcu eleifend pretium nec sed tortor. Cras tristique ex quis aliquam vulputate.",
	read: true,
};

const toggleReadStatus = () => {
	note.read = !note.read;
};

it("note object checks matchers and pass", () => {
	expect(note).toMatchSnapshot({
		subject: expect.any(String),
		body: expect.any(String),
		read: expect.any(Boolean),
	});
});

it("renders correctly", () => {
	const tree = create(
		<NoteDetails note={note} toggleReadStatus={toggleReadStatus} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it("instance is null when note prop is empty object", () => {
	const emptyNote = {};
	let component;
	act(() => {
		component = create(
			<NoteDetails note={emptyNote} toggleReadStatus={toggleReadStatus} />
		);
	});
	component.toJSON();
	const instance = component.getInstance();
	expect(component).toMatchSnapshot();
	expect(instance).toBe(null);
});

it("toggles the read status when mark as read/unread button clicked", () => {
	let component;
	act(() => {
		component = create(
			<NoteDetails note={note} toggleReadStatus={toggleReadStatus} />
		);
	});

	const read = note.read;

	const instance = component.root;
	const button = instance.findByType("button");
	const buttonInnerText = button.props.children;

	act(() => {
		button.props.onClick();
	});
	expect(note.read).toBe(!read);
	expect(buttonInnerText).toBe("Mark as unread");
});
