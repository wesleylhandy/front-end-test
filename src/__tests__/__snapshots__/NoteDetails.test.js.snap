// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
Object {
  "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non velit vehicula lacus dictum tristique. Quisque ac justo magna. Duis eu est purus. Quisque ut enim tortor. Morbi interdum risus a lorem posuere dapibus. Morbi orci est, varius sed dolor quis, efficitur scelerisque sapien. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Aenean at libero sapien. Morbi ultrices malesuada nisl, sit amet aliquam ipsum egestas non. Nunc at massa vel metus vestibulum fringilla in vitae est. Pellentesque eget lorem eget arcu eleifend pretium nec sed tortor. Cras tristique ex quis aliquam vulputate.",
  "read": true,
  "subject": "Lorem ipsum dolor sit amet",
}
`;

exports[`instance is null when note prop is empty object 1`] = `null`;

exports[`renders correctly 1`] = `
<section
  className="NoteDetails"
>
  <h3
    className="NoteDetails__title"
  >
    Lorem ipsum dolor sit amet
  </h3>
  <p
    className="NoteDetails__subject"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non velit vehicula lacus dictum tristique. Quisque ac justo magna. Duis eu est purus. Quisque ut enim tortor. Morbi interdum risus a lorem posuere dapibus. Morbi orci est, varius sed dolor quis, efficitur scelerisque sapien. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Aenean at libero sapien. Morbi ultrices malesuada nisl, sit amet aliquam ipsum egestas non. Nunc at massa vel metus vestibulum fringilla in vitae est. Pellentesque eget lorem eget arcu eleifend pretium nec sed tortor. Cras tristique ex quis aliquam vulputate.
  </p>
  <button
    className="Button"
    onClick={[Function]}
    style={Object {}}
  >
    Mark as unread
  </button>
</section>
`;
